resource_estimator/src/counts/tests.rs

```diff
@@ -12,6 +12,7 @@ use qsc::{
     target::Profile,
     PackageType, SourceMap,
 };
+use qsc_data_structures::language_features::LanguageFeatures;
```

Comment:
I do not think that we should introduce references to compiler into the resource_estimator. In an ideal case, RE should be independent from the compiler and specific language. So, I would prefer to avoid this. However, if there is a cascade referencing requiring this, then OK - let us proceed.

Reply:
Sounds good. We may consider separating RE from the qsharp packaging since it could support other languages and should not depend on qsharp. So, if there is anything we could do to prevent potential issues with this, it would be better to take it into account.

Reply:
It's also worth noting that this part of the resource estimator is the part that connects to the evaluator and handles gathering logical counts during simulation. It requires a tight connection to the Q# infrastructure to accomplish that. We could separate that into its own crate that depends on the physical estimator and qsc, such that the physical estimator would not depend on another code in the repo. Some of the original designs had such a division, we just haven't had the chance to do that refactoring yet.